Documentation/design-docs/multilevel-sharedfx-lookup.md

```diff
@@ -145,22 +147,46 @@ So, for example, if there are two references:
-  - One potential future case where is makes sense to have "pseudo-circular" dependencies is when framework "foo" loads a light-up framework which depends on "foo". Internally the foo->lightup reference may be treated as a late-bound framework reference, thus causing a cycle. This potential feature may replace the "additional deps" feature in a way that allows for richer light-up scenarios by allowing the lightup to specify framework dependency(s) and have a small deps.json.
+  - One potential future case is to allow "pseudo-circular" dependencies where framework "foo" loads a light-up framework which depends on "foo". Internally the foo->lightup reference may be treated as a late-bound framework reference, thus causing a cycle. This potential feature may replace the "additional deps" feature in a way that allows for richer light-up scenarios by allowing the lightup to specify framework dependency(s) and have a small deps.json.
 - <B>No Downgrading:</B> a newer version of a shared framework should keep or increase the version to another shared framework (never decrease the version number).
-  
 By following these best practices we have optimal run-time performance (less processing and probing) and less chance of incompatible framework references.
 
-Discussion points:
+#### Algorithm
+Terminology:
+- `config list`: entries for a single runtimeconfig.json which consists of framework `name`, `version`, optional `applyPatches`, and optional `rollForwardOnNoCandidateFx`.
+- `newest list`: entries keyed off of framework name that contain the highest framework version requested. It is used to perform "soft" roll-forwards to compatible references of the same framework name without reading the disk or performing excessive re-try (Step 7).
+- `resolved list`: a list of frameworks that have been resolved, meaning a compatible framework was found on disk.
+
+Algorithm:
+1. Determine the `config list`:
+  - Parse the application's runtimeconfig.json `runtimeOptions.frameworks` section.
+  - If the `runtimeOptions.framework.name` and `runtimeOptions.framework.version` exist, Then insert that framework into the beginning of the `config list`.
+1. For each framework in `config list`:
+1. --> If the framework is not currently in the `newest map` list Then add it.
+  - By doing this here, before the next loop, we minimize the number of re-try attempts.
+1. For each framework in `config list`:
+1. --> If the framework is not in `resolved list` Then resolve the framework
+ - Update `newest list` if current version is newer
+ - We may fail here if not compatible.
+ - Probe for the framework on disk
+ - If success add it to `resolved list` and make a recursive call back to Step 2 with but pass in a new `config list` based upon the values from the newly resolved framework's runtimeconfig.json which may reference additional frameworks.
```

Comment:
For me the numbering shows wrong - so it's hard to tell which one is Step 2.

Reply:
Will fix with hard-coded numbers; was showing correctly in Atom markdown editor prior